hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -240,8 +247,19 @@ public VersionResponse getVersion(SCMVersionRequestProto versionRequest) {
   @Override
   public RegisteredCommand register(
       DatanodeDetails datanodeDetails, NodeReportProto nodeReport,
-      PipelineReportsProto pipelineReportsProto) {
-
+      PipelineReportsProto pipelineReportsProto,
+      LayoutVersionProto layoutInfo) {
+
+    if (layoutInfo != null) {
+      if (layoutInfo.getSoftwareLayoutVersion() >
+          scmLayoutVersionManager.getSoftwareLayoutVersion()) {
```

Comment:
For another case, if register dn layout version lower than scm version, do we need to print some log here that can be helpful for us know some old version DN?

Reply:
SCM can handle multiple DNs and it may make log verbose. This will be thrown back to the DN and it will be logged there for debugging as to why datanode is not able to register with SCM.

Reply:
Okay, makes sense to me.